### PR TITLE
Horizontal table of contents implementation

### PR DESCRIPTION
**Pull request by KashishMistry**
### Related Item(s)
#402 

### Changes
- Added horizontal table of contents component
- Implementation is such that the config of the project metadata will determine if the table of contents will be horizontal or vertical
- Updated version to `3.1.2` in `package.json`

### Notes
horizontaltabs

### Testing
With this PR, the horizontal table of contents will be shown in the demo (the vertical table of contents has been commented out along with some styles, will be removed before merging) to test the component.

- - -
This change is [Reviewable](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413)

### CODE REVIEW

**Review by yileifeng — CHANGES_REQUESTED** [review]
Looks good, just a question but do we want to have the horizontal ToC collapsible? 

Just as a reminder will also need to update the package.json version and publish to NPM once everything is tested and working for the editor preview. 

> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413#-:-Nt1ZAWw05ziiNyKRdBn:bbz451f)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @KashishMistry)

___
*[`src/components/story/story-content.vue` line 4 at r1](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413#-Nt1Z1zd1s7Fm8ZavFJA:-Nt1Z1zd1s7Fm8ZavFJB:b-2dmhfs) ([raw file](https://github.com/ramp4-pcar4/story-ramp/blob/bb1b2fd7fd7c5647dba9ee7381fd11be037193a9/src/components/story/story-content.vue#L4)):*
> ```Vue
>     <div class="items-stretch">
>         
>         
> ```

Blocking until ready to merge

**Review by RyanCoulsonCA — CHANGES_REQUESTED** [review]
Awesome work, looks great! I noticed that when you click on one of the chapters part of the content is blocked by the ToC. We may need to tweak the `scrollToChapter` function to account for the height of the ToC element to get around this.

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413#-:-Nt1a33I9xQEk-cS2HYL:b1qzanm)* status: all files reviewed, 1 unresolved discussion (waiting on @KashishMistry)

**Review by KashishMistry — COMMENTED** [review]
Donethanks (for the scrolling issue).

> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413#-:-NtSTSaO4PMiZQxrkunZ:b-bryl2j)* status: 3 of 4 files reviewed, 1 unresolved discussion (waiting on @RyanCoulsonCA)

**Review by RyanCoulsonCA — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413)* status: all files reviewed, 1 unresolved discussion (waiting on @KashishMistry)

**Comment by KashishMistry** [comment]
Some formatting changes made to align table of contents and set css properties dynamically

**Review by yileifeng — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 3 of 4 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @KashishMistry)

**Review by yileifeng — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @KashishMistry)

**Review by dnlnashed — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 3 of 4 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @KashishMistry)

**Review by yileifeng — APPROVED** [review]
> Reviewed 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @KashishMistry)

**Review by yileifeng — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413)* status: all files reviewed, 1 unresolved discussion (waiting on @KashishMistry)

___
*[`src/components/story/story-content.vue` line 3 at r7](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413#-Nu0H_O4EekSjjKZ3MNV:-Nu0H_O539r0f_B3wiiW:b9a6as0) ([raw file](https://github.com/ramp4-pcar4/story-ramp/blob/91011ca53f6222fc31ac6c7788d5bcc92e636c08/src/components/story/story-content.vue#L3)):*
> ```Vue
> <template>
>     <div class="items-stretch">
>         
> ```

blocking until merge

**Review by RyanCoulsonCA — APPROVED** [review]
> Reviewed 2 of 4 files at r4, 1 of 1 files at r6, 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413)* status: all files reviewed, 1 unresolved discussion (waiting on @KashishMistry)

**Review by yileifeng — APPROVED** [review]
> Reviewed 3 of 3 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/story-ramp/413)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @KashishMistry)

**Pull request merged**
